src/sdks/core/src/cpp/templates/Lifecycle/src/module_impl.cpp

```diff
@@ -0,0 +1,152 @@
+/* Copyright 2023 Comcast Cable Communications Management, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+
+#include "${info.title.lowercase}_impl.h"
+
+${if.implementations}
+namespace Firebolt {
+namespace ${info.Title} {
+${if.providers}
+/* ${PROVIDERS} */${end.if.providers}
+
+EnumMap<Firebolt::Lifecycle::LifecycleState> lifecycleStateMap = {
+    { Firebolt::Lifecycle::LifecycleState::INITIALIZING, "initializing" },
+    { Firebolt::Lifecycle::LifecycleState::INACTIVE, "inactive" },
+    { Firebolt::Lifecycle::LifecycleState::FOREGROUND, "foreground" },
+    { Firebolt::Lifecycle::LifecycleState::BACKGROUND, "background" },
+    { Firebolt::Lifecycle::LifecycleState::UNLOADING, "unloading" },
+    { Firebolt::Lifecycle::LifecycleState::SUSPENDED, "suspended" }
+};
+
+
+/* ready - Notify the platform that the app is ready */
+static void readyDispatcher(const void* result) {
+    Firebolt::IFireboltAccessor::Instance().MetricsInterface().ready();
+}
+
+// localCallback to update the state
+static void onReadyInnerCallback(void* notification, const void* userData, void* jsonResponse )
+{
+    const LifecycleImpl* selfConst = static_cast<const LifecycleImpl*>(userData);
+    LifecycleImpl* self = const_cast<LifecycleImpl*>(selfConst);
+
+    WPEFramework::Core::ProxyType<JsonData_LifecycleEvent>& proxyResponse = *(reinterpret_cast<WPEFramework::Core::ProxyType<JsonData_LifecycleEvent>*>(jsonResponse));
+    ASSERT(proxyResponse.IsValid() == true);
+
+    if (proxyResponse.IsValid() == true) {
+        LifecycleEvent value;
+
+        value.state = proxyResponse->State;
+        std::string stateStr = ConvertEnum<Firebolt::Lifecycle::LifecycleState>(lifecycleStateMap, value.state);
+        // Assign stateStr to currentState in ${info.Title}Impl instance
+        self->currentState = stateStr;
+        std::cout << "Updated the Current State to: " << self->currentState << std::endl;
+
+        proxyResponse.Release();
+
+    }
+}
+ 
+
+
+/* ready - Notify the platform that the app is ready */
+void ${info.Title}Impl::ready(Firebolt::Error *err) {  
+    Firebolt::Error status = Firebolt::Error::NotConnected;
+
+    JsonObject jsonParameters;
+
+    // Call Prioritize with  callback
+    status = FireboltSDK::Event::Instance().Prioritize<JsonData_LifecycleEvent>("lifecycle.onForeground", jsonParameters, onReadyInnerCallback, (void*)nullptr, this);
```

Comment:
This will call the priortize function,that will subscribe to the foreground event and get added to the internalMap.Later at the time of dispatch internal events will get first priority and executes its callback